src/index.ts

```diff
@@ -264,8 +264,13 @@ export async function generate (config: OpenAPIConfiguration) {
         })
       }
     }),
-    schemas: spec.components!.schemas
+    schemas: spec.components!.schemas,
+    middlewares: Object
+      .values(codegenControllers)
+      .flatMap(controller => controller.flatMap(method => method.middlewares || []))
+      .filter((middleware, index, self) => self.findIndex(m => m.name === middleware.name) === index)
```

Comment:
I have weird feelings about this approach. I could implement importing the middlewares in the template file in loop `{{#each controllers}}`, but that would add complexity to the template, which I want to avoid since other templates would also need to implement it.